lib/vagrant/environment.rb

```diff
@@ -564,6 +565,11 @@ def push(name)
           pushes: push_registry.keys
       end
 
+      # Validate the global push configuration and our local push configuration
+      machine = self.machine(self.machine_names.first, self.default_provider)
+      machine.action_raw(:config_validate, Vagrant::Action::Builtin::ConfigValidate)
+      config.validate(machine)
```

Comment:
@mitchellh I looked that that, but there's no `vm` in our case, since this only depends on the environment, so that doesn't trigger (at least it wasn't in my tests). I'm not sure how this should all be wired together since the config validation logic is a bit confusing to me, but this is definitely the bug.